### PR TITLE
[MNOE-155] - Staff invite workflow

### PR DESCRIPTION
**Pull request by clemthenem**
- Fixed bugs:
  * Staff did not received a confirmation email
  * Reloading the staff page was not working

- Improvments:
  * Redirect an "admin" user when he changes its details to "staff" to avoid accessing the staff page
  * Stop the last admin user to change its status or delete himself (frontend check)
  * Exit button logout staff without organizations and redirect to the dashboard otherwise.

### CODE REVIEW

**Comment by clemthenem** [comment]
@ouranos @alexnoox Can you review this? :)

File: frontend-admin/src/app/components/mnoe-api/admin/current-user.svc.coffee [hunk 1]
```diff
@@ -17,7 +17,7 @@
   # Save the current user in variable to be able to reference it directly
   @user = {}
 
-  # Get the current user admmin role
+  # Get the current user admin role
   @getAdminRole = ->
```
Comment (alexnoox) [inline frontend-admin/src/app/components/mnoe-api/admin/current-user.svc.coffee]: While we're at it:
- Can you rename this function `getUser()`
- Assign (or `angular.copy` if needed) the response to `_self.user` directly, not only the role

File: frontend-admin/src/app/components/mnoe-api/admin/current-user.svc.coffee [hunk 2]
```diff
@@ -28,13 +28,32 @@
...
+        $timeout(->
+          # Runs after the authentication promise has been rejected.
+          $state.go('dashboard.home')
+        )
+        deferred.reject()
+    )
+    return deferred
+
+  @logoutIfNoOrganisation = ->
+    userPromise.then(
```
Comment (alexnoox) [inline frontend-admin/src/app/components/mnoe-api/admin/current-user.svc.coffee]: use `_self.getUser()`

File: frontend-admin/src/app/components/mnoe-api/admin/current-user.svc.coffee [hunk 3]
```diff
@@ -28,13 +28,32 @@
...
+        if response.data.organizations? && response.data.organizations.length > 0
+          $window.location.assign("/dashboard/")
+        # Logout if the user has no organization
+        else
+          $window.location.assign("/dashboard/#/logout")
+    )
+
+  @updateAdminRole = ->
+    userPromise = null
+    _self.skipIfNotAdmin()
```
Comment (alexnoox) [inline frontend-admin/src/app/components/mnoe-api/admin/current-user.svc.coffee]: I think you should then call `_self.getUser()` now here to refresh the user. Call this function `refreshUser()`

File: frontend-admin/src/app/components/mnoe-api/admin/users.svc.coffee [hunk 4]
```diff
@@ -68,6 +68,11 @@
   @sendSignupEmail = (email) ->
     MnoeAdminApiSvc.all('/users').doPOST({user: {email: email}}, 'signup_email')
 
+  # Send an email to a user with the link to the confirmation page
+  # POST /mnoe/jpi/v1/admin/users/invites
+  @sendConfirmationEmail = (email) ->
```
Comment (alexnoox) [inline frontend-admin/src/app/components/mnoe-api/admin/users.svc.coffee]: Is it used?

File: frontend-admin/src/app/services/mnoe-observables/mnoe-observables.coffee [hunk 5]
```diff
@@ -14,7 +14,10 @@
       callback(object)
 
   @unsubscribe = (name, cb) ->
-    _.remove(observersCallbacks[name], cb) ->
-      delete observersCallbacks[name] if _.isEmpty(observersCallbacks[name])
+    _.pull(observersCallbacks[name], cb)
+    delete observersCallbacks[name] if _.isEmpty(observersCallbacks[name])
+
+  @print =  ->
+    observersCallbacks
```
Comment (alexnoox) [inline frontend-admin/src/app/services/mnoe-observables/mnoe-observables.coffee]: Remove this method, it is only used for testing purpose.

File: frontend-admin/src/app/views/customers/customers.controller.coffee [hunk 6]
```diff
@@ -1,4 +1,4 @@
-@App.controller 'CustomersController', ($uibModal, MnoeUsers, MnoeOrganizations, MnoeObservables, OBS_KEYS) ->
+@App.controller 'CustomersController', ($uibModal, $scope, MnoeUsers, MnoeOrganizations, MnoeObservables, OBS_KEYS) ->
```
Comment (alexnoox) [inline frontend-admin/src/app/views/customers/customers.controller.coffee]: Plese use this order: `$scope, $uibModal, MnoeUsers, MnoeOrganizations, MnoeObservables, OBS_KEYS`

File: frontend-admin/src/app/views/customers/customers.controller.coffee [hunk 7]
```diff
@@ -14,12 +14,11 @@
       controllerAs: 'vm'
     )
 
-  MnoeObservables.registerCb(OBS_KEYS.userChanged, (promise) ->
-    promise.then(
-      (response) ->
-        vm.users.totalCount = response.headers('x-total-count')
-      )
-  )
+  processListOfUsers = (response) ->
```
Comment (alexnoox) [inline frontend-admin/src/app/views/customers/customers.controller.coffee]: Please use a more meaningful name `updateUsersCounter` or `updateUsersKPI`

File: frontend-admin/src/app/components/mnoe-api/admin/current-user.svc.coffee (hunk 2 above)
Comment (alexnoox) [inline frontend-admin/src/app/components/mnoe-api/admin/current-user.svc.coffee]: Just call this function `logout`

**Comment by clemthenem** [comment]
Thanks @alexnoox  for your comments! Let me know if I need to change anything else

**Comment by alexnoox** [comment]
LGTM

@ouranos Could you review this one too? Cheers

**Comment by clemthenem** [comment]
@ouranos what do you think of this one?

**Comment by clemthenem** [comment]
@ouranos This should fix Malcom's questions about adding admins

**Comment by clemthenem** [comment]
@ouranos @alexnoox can we merge this before I add the rewrite of the frontend?

**Comment by clemthenem** [comment]
I splitted this PR in two seperate ones:
- https://github.com/maestrano/mnoe-admin-panel/pull/6 for the frontend
- https://github.com/maestrano/mno-enterprise/pull/232 for the backend

**Comment by clemthenem** [comment]
@alexnoox this is the PR for the frontend-admin

**Pull request merged**